docs/query-builder.txt

```diff
@@ -9,588 +9,1095 @@ Query Builder
 
-**Like**
+.. _laravel-query-builder-exists:
 
-.. code-block:: php
+Contains a Field Example
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-   $spamComments = Comment::where('body', 'like', '%spam%')->get();
+The following example shows how to use the ``exists()``
+query builder method to match documents that contain the
+field ``random_review``:
 
-**Aggregation**
+.. literalinclude:: /includes/query-builder/MovieController.php
+   :language: php
+   :dedent:
+   :start-after: begin query exists
+   :end-before: end query exists
 
-**Aggregations are only available for MongoDB versions greater than 2.2.x**
+To learn more about this query operator, see
+:manual:`$exists </reference/operator/query/exists/>`
+in the {+server-docs-name+}.
 
-.. code-block:: php
+.. _laravel-query-builder-all:
 
-   $total = Product::count();
-   $price = Product::max('price');
-   $price = Product::min('price');
-   $price = Product::avg('price');
-   $total = Product::sum('price');
+Contains All Fields Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Aggregations can be combined with **where**:
+The following example shows how to use the ``all`` query
+operator with the ``where()`` query builder method to match
+documents that contain all the specified fields:
 
-.. code-block:: php
+.. literalinclude:: /includes/query-builder/MovieController.php
+   :language: php
+   :dedent:
+   :start-after: begin query all
+   :end-before: end query all
 
-   $sold = Orders::where('sold', true)->sum('price');
+To learn more about this query operator, see
+:manual:`$all </reference/operator/query/all/>`
+in the {+server-docs-name+}.
 
-Aggregations can be also used on sub-documents:
+.. _laravel-query-builder-size:
 
-.. code-block:: php
+Match Array Size Example
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-   $total = Order::max('suborder.price');
+The following example shows how to pass the ``size``
+query operator with the ``where()`` query builder
+method to match documents that contain a ``directors``
+field that contains an array of exactly five elements:
 
-.. note::
+.. literalinclude:: /includes/query-builder/MovieController.php
+   :language: php
+   :dedent:
+   :start-after: begin query size
+   :end-before: end query size
 
-   This aggregation only works with single sub-documents (like ``EmbedsOne``)
-   not subdocument arrays (like ``EmbedsMany``).
+To learn more about this query operator, see
+:manual:`$size </reference/operator/query/size/>`
+in the {+server-docs-name+}.
 
-**Incrementing/Decrementing the value of a column**
+.. _laravel-query-builder-type:
 
-Perform increments or decrements (default 1) on specified attributes:
+Match Data Type Example
+~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: php
+The following example shows how to pass the ``type``
+query operator with the ``where()`` query builder
+method to match documents that contain a ``directors``
+field that contains an array of exactly five elements:
```

Comment:
I: description and example do not demonstrate the functionality described in this section for type

Reply:
Good catch, thanks!